src/components/builder/validate/soft-required.tsx

```diff
@@ -0,0 +1,51 @@
+import {useFormikContext} from 'formik';
+import {useEffect} from 'react';
+import {FormattedMessage, useIntl} from 'react-intl';
+
+import {Checkbox} from '../../formio';
+
+type ComponentWithRequiredOptions = {
+  validate?: {
+    required?: boolean;
+  };
+  openForms?: {
+    softRequired?: boolean;
+  };
+};
+
+const SoftRequired = () => {
+  const intl = useIntl();
+  const {values, setFieldValue} = useFormikContext<ComponentWithRequiredOptions>();
+
+  const isRequired = values?.validate?.required ?? false;
+  const isSoftRequired = values?.openForms?.softRequired ?? false;
+
+  // if the field is hard required, we must disable the soft required option, and
+  // synchronize the softRequired to uncheck the option (if needed)
+  useEffect(() => {
+    if (!(isRequired && isSoftRequired)) return;
+    setFieldValue('openForms.softRequired', false);
```

Comment:
It's a bit complicated... Maybe this will be simpler:
```suggestion
    if (isRequired && isSoftRequired) {
      setFieldValue('openForms.softRequired', false);
    }
```

Reply:
yeah I'm a big fan of guard clauses myself, but this reads better :)